esmvaltool/diag_scripts/ocean/diagnostic_maps_multimodel.py

```diff
@@ -0,0 +1,316 @@
+from esmvaltool.diag_scripts.shared import run_diagnostic
+
+# This part sends debug statements to stdout
+logger = logging.getLogger(os.path.basename(__file__))
+logging.getLogger().addHandler(logging.StreamHandler(sys.stdout))
+
+
+def add_map_plot(ax,
+                 cube,
+                 nspace,
+                 cols,
+                 title='',
+                 cmap='',
+                 extend='neither',
+                 hascbar=False):
+    """
+    Add a map in the current pyplot suplot.
+
+    Parameters
+    ----------
+    ax: object
+        The matplotlib.pyplot Axes object
+    cube: iris.cube.Cube
+        the iris cube to be plotted.
+    nspace: numpy.array
+        An array of the ticks of the colour part.
+    cols: integer
+        Number of columns in the multipanel plot
+    title: str
+        A string to set as the subplot title.
+    cmap: str
+        A string to describe the matplotlib colour map.
+    extend: str
+        Contourf-coloring of values outside the levels range
+    hascbar: logical
+        Add colorbar to the subplot
+    """
+    iris.plot.contourf(
+        cube,
+        nspace,
+        linewidth=0,
+        cmap=plt.cm.get_cmap(cmap),
+        extend=extend,
+        zmin=nspace.min(),
+        zmax=nspace.max())
+
+    ax.coastlines()
+    gls = ax.gridlines(draw_labels=False, color='black', alpha=0.4)
+    gls.ylocator = mticker.FixedLocator(np.linspace(-90., 90., 7))
+    ax.set_title(title, fontweight="bold", fontsize='large')
+
+    if hascbar:
+        bba = (0., -0.1, 1, 1)
+        if cols > 0:
+            bba = ((0.5 - cols) * 1.1, -0.15, cols, 1)
+        axins = inset_axes(
+            ax,
+            width="95%",
+            height="6%",
+            loc='lower center',
+            bbox_to_anchor=bba,
+            bbox_transform=ax.transAxes,
+            borderpad=0,
+        )
+        cbar = plt.colorbar(orientation='horizontal', cax=axins)
+        cbar.set_ticks(nspace[::2])
+
+
+def make_multiple_plots(cfg, metadata, obs_filename):
+
+    """
+    Produce multiple panel comparison maps of model(s) and data (if provided).
+    If observations are not provided, plots of each model data are drawn.
+
+    Put on top row observational data (if available) and in following subplots
+    model difference (or data) organized in rows/cols using row/col layout.
+
+    Parameters
+    ----------
+    cfg: dict
+        the opened global config dictionary, passed by ESMValTool.
+    metadata: dict
+        the input files dictionairy
+    obs_filename: str
+        the preprocessed observations file.
+    """
+    logger.debug('make_multiple_plots')
+    # ####
+    filenames = list(metadata.keys())
+    proj = ccrs.Robinson(central_longitude=0)
+    # plot layout
+    layout = metadata[filenames[0]]['layout_rowcol']
+
+    # check if observations are provided
+    hasobs = False
+    obsname = ''
+    if obs_filename != '':
```

Comment:
can initialize it with `None` rather than an empty string, hence make it an optional argument in the func

Reply:
@valeriupredoi wrt to `obsname` I would prefer to keep it as an empty string so it will make less complex to handle the filename generation at the end ...
https://github.com/ESMValGroup/ESMValTool/blob/a3390e6ece121ac0c97f571b0955fc21e0cbb4ec/esmvaltool/diag_scripts/ocean/diagnostic_maps_multimodel.py#L257-L259

Reply:
okey man